### PR TITLE
doc: add description for action arguments (verify and side_effect)

### LINKED ISSUE #3591: No doc for #3467
The new feature added in #3467 is not documented in the docs. 

The new feature:

> provide[s] alternative location for
> tests or side effects playbook in test sequence like this:
> 
>  ```
>  test_sequence:
>     - create
>     - converge
>     - verify
>     - side_effect side_effect1.yml
>     - verify tests_for_effect1/
>     - side_effect side_effect2.yml
>     - verify tests_for_effect2/
>  ```

### PR DESCRIPTION
**Pull request by amarao**
Adding docs for action arguments.

Should fix #3591

I'm not familiar with syntax and hope it's done right.

### CODE REVIEW

File: src/molecule/scenario.py [hunk 1]
```diff
@@ -88,6 +88,66 @@ class Scenario(object):
...
+          - side_effect reboot.yaml
+          - verify after_reboot/
+          - side_effect alter_configs.yaml
+          - converge
+          - verify other_test1.py other_test2.py
+          - side_effect
+          - verify
+
+    ``side_effect`` without an argument is executing usual `side_effect` configured in
+      provisioner.playbooks section of molecule.yml.
```
Comment (ziegenberg) [inline src/molecule/scenario.py:113]:
```suggestion
    ``side_effect`` without an argument is executing the usual `side_effect` configured in
      `provisioner.playbooks` section of molecule.yml.
```

**Review by ziegenberg — CHANGES_REQUESTED** [review]
Hi @amarao!

Thank you very much for your contribution; it's much appreciated!
I applied the labels, so now only one more CI check is red. Can you fix this? Also, some minor suggestions for grammar improvements.

**Comment by amarao** [comment]
@ziegenberg Thank you a lot for suggestions. (I don't have GH permissions to 'Commit suggestions', so I applied them manually). I believe I fixed `tox -e docs` part of pipeline.

**Comment by ziegenberg** [comment]
Looks good now!

File: src/molecule/scenario.py [hunk 2]
```diff
@@ -88,6 +88,70 @@ class Scenario(object):
...
+    ``verify`` without an argument is executing usual tests configured in the verifier section
+    of molecule.yml.
+
+    If one or more arguments (separated by spaces) are present, each argument is treated
+    as a test name (file or directory) to pass to the verifier (either Ansible or Testinfra).
+    The kind of verifier is set in the `verifier` section of molecule.yml and is applied to all
+    `verify` actions in the scenario.
+
+    The path to tests is relative to the molecule.yml file location. The `additional_files_or_dirs`
+    setting for verifier is ignore if the `verify` action has an argument.
```
Comment (ziegenberg) [inline src/molecule/scenario.py:131]:
```suggestion
    setting for verifier is ignored if the `verify` action has an argument.
```

**Review by ziegenberg — CHANGES_REQUESTED** [review]
One more typo and we're good.

File: src/molecule/scenario.py (hunk 2 above)
Comment (amarao) [reply src/molecule/scenario.py]: Thanks! This one and one clumsy thing at the beginning fixed.

**Review by ziegenberg — APPROVED** [review]

**Pull request merged**